ds248x/ds248x.go

```diff
@@ -154,6 +170,73 @@ func (d *Dev) Search(alarmOnly bool) ([]onewire.Address, error) {
 	return onewire.Search(d, alarmOnly)
 }
 
+// ChannelSelect function is for selecting one of eight 1-w channels on DS2482-800.
+// On other chips it does nothing. Channel selection values are between
+// 0 and 7. It is expected that application keeps track of
+// with 1-w device is connected to with channel.
+// Communication error is returned if present.
+func (d *Dev) ChannelSelect(ch int) error {
+	var err error = nil
+	switch d.isDS248x {
+	case isDS2482x100:
+		// if ch != 0 {
+		// 	err = errors.New("channel != 0")
+		// 	return fmt.Errorf("ds2482-100: error while selecting channel: %s", err)
+		// }
+	case isDS2482x800:
+		if ch < 0 {
+			err = errors.New("channel < 0")
```

Comment:
I wouldn't bother with different error string for <0 or >7. Also why create two errors?